BasicGates/BasicGates.ipynb

```diff
@@ -546,7 +546,52 @@
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "### Task 2.4. Toffoli gate\n",
+    "### Task 2.4. Two-qubit gate - 4\n",
+    "\n",
+    "**Input:** Two unentangled qubits (stored in an array of length 2) in an arbitrary two-qubit state ${\\color{blue}\\alpha} |00\\rangle + {\\color{blue}\\beta} |01\\rangle + \\gamma |10\\rangle + \\delta |11\\rangle$.\n",
+    "\n",
+    "\n",
+    "**Goal:**  Change the two-qubit state to ${\\color{red}\\beta} |00\\rangle + {\\color{red}\\alpha} |01\\rangle + \\gamma |10\\rangle + \\delta |11\\rangle$.\n",
+    "\n",
+    "> This task can be solved using one library function; as an exercise, try to express the solution using several (possibly controlled) Pauli gates.  \n",
+    ">\n",
+    "> If you can't come up with a solution, or if you have solved the task, read about the relevant library function: \n",
+    "  <details> \n",
+    "    <summary><b>Click here.</b></summary>\n",
+    "    `ControlledOnInt` function can generate a controlled gate, which is applied only when the control qubit is in the $|0\\rangle$ state. </br>\n",
+    "    For example, `ControlledOnInt(0, H)([qs[0]], qs[1])` applies the Hadamard gate to the `qs[1]` qubit, when the control qubit `qs[0]` is in the $|0\\rangle$ state.\n",
```

Comment:
I liked your explaination for `ControlledOnInt` function in https://github.com/microsoft/QuantumKatas/pull/824. 

This might create a misunderstanding because we are saying that  `ControlledOnInt` function is only applied when register is in $| 0 \rangle$ state. This is  not always true because I can pass one to `ControlledOnInt` function, and then this gate will be applied if control register is in state $|1\rangle$

```suggestion
    "    `ControlledOnInt` function can generate a controlled gate, which is applied when the control register is in a state corresponding to a non-negative integer</br>\n",
    "    Example 1: For integer 0,  `ControlledOnInt(0, H)([qs[0]], qs[1])` applies the Hadamard gate to the `qs[1]` qubit, when the control qubit `qs[0]` is in the $|0\\rangle$ state.<br>\n",
    "    Example 2: For integer 0,  `ControlledOnInt(0, H)([qs[0]], qs[1])` applies the Hadamard gate to the `qs[1]` qubit, when the control qubit `qs[0]` is in the $|0\\rangle$ state.<br>\n",
```

Reply:
This is exactly why I've written "**can** generate" instead of "generates", but I see now that it generates confusion. Your version is better, but there are still some improvements needed (as both examples are the same).

Reply:
Thanks for catching this

Reply:
Could you just wait a minute? I am editing my suggestion so that you can apply it.Thanks.

Reply:
Ahh, I didn't expect such a quick reaction. I already applied the previous version :/

But I also wanted to reword one of your other suggestions, so I'm doing changes anyway

Reply:
That was in a different file

Reply:
I have edited my suggestion so that you can apply it.

Reply:
All of the relevant changes should be in [this commit] (https://github.com/microsoft/QuantumKatas/pull/827/commits/3473dabb461ef5a58b68c59f7662f9bfe2378a37)